validator/accounts/accounts_enable_disable.go

```diff
@@ -0,0 +1,273 @@
+			)
+			if err != nil {
+				return err
+			}
+			if strings.ToLower(resp) == "n" {
+				return nil
+			}
+		} else {
+			promptText := "Are you sure you want to disable %d accounts? (%s) Y/N"
+			if len(filteredPubKeys) == len(validatingPublicKeys) {
+				promptText = fmt.Sprintf("Are you sure you want to disable all accounts? Y/N (%s)", au.BrightGreen(allAccountStr))
+			} else {
+				promptText = fmt.Sprintf(promptText, len(filteredPubKeys), au.BrightGreen(allAccountStr))
+			}
+			resp, err := promptutil.ValidatePrompt(os.Stdin, promptText, promptutil.ValidateYesOrNo)
+			if err != nil {
+				return err
+			}
+			if strings.ToLower(resp) == "n" {
+				return nil
+			}
+		}
+	}
+	if err := DisableAccounts(cliCtx.Context, &AccountsConfig{
+		Wallet:            w,
+		Keymanager:        keymanager,
+		DisablePublicKeys: rawPublicKeys,
+	}); err != nil {
+		return err
+	}
+	log.WithField("publicKeys", allAccountStr).Info("Accounts disabled")
+	return nil
+}
+
+func EnableAccountsCli(cliCtx *cli.Context) error {
+	w, err := wallet.OpenWalletOrElseCli(cliCtx, func(cliCtx *cli.Context) (*wallet.Wallet, error) {
+		return nil, wallet.ErrNoWalletFound
+	})
+	if err != nil {
+		return errors.Wrap(err, "could not open wallet")
+	}
+	ikeymanager, err := w.InitializeKeymanager(cliCtx.Context, &iface.InitializeKeymanagerConfig{
+		SkipMnemonicConfirm: false,
+	})
+	if err != nil {
+		return errors.Wrap(err, "could not initialize keymanager")
+	}
+	switch w.KeymanagerKind() {
+	case keymanager.Remote:
+		return errors.New("cannot enable accounts for a remote keymanager")
+	case keymanager.Imported:
+		km, ok := ikeymanager.(*imported.Keymanager)
+		if !ok {
+			return errors.New("not a imported keymanager")
+		}
+		disabledPublicKeys := km.KeymanagerOpts().DisabledPublicKeys
+		if len(disabledPublicKeys) == 0 {
+			return errors.New("No accounts are disabled.")
+		}
+		disabledPublicKeys48 := make([][48]byte, len(disabledPublicKeys))
+		for i := range disabledPublicKeys {
+			disabledPublicKeys48[i] = bytesutil.ToBytes48(disabledPublicKeys[i])
+		}
+
+		// Allow the user to interactively select the accounts to enable or optionally
+		// provide them via cli flags as a string of comma-separated, hex strings.
+		filteredPubKeys, err := filterPublicKeysFromUserInput(
+			cliCtx,
+			flags.EnablePublicKeysFlag,
+			disabledPublicKeys48,
+			prompt.SelectAccountsEnablePromptText,
+		)
+		if err != nil {
+			return errors.Wrap(err, "could not filter public keys for activation")
+		}
+		rawPublicKeys := make([][]byte, len(filteredPubKeys))
+		formattedPubKeys := make([]string, len(filteredPubKeys))
+		for i, pk := range filteredPubKeys {
+			pubKeyBytes := pk.Marshal()
+			rawPublicKeys[i] = pubKeyBytes
+			formattedPubKeys[i] = fmt.Sprintf("%#x", bytesutil.Trunc(pubKeyBytes))
+		}
+		allAccountStr := strings.Join(formattedPubKeys, ", ")
+		if !cliCtx.IsSet(flags.DisablePublicKeysFlag.Name) {
```

Comment:
Is this correct?

Reply:
good catch.. strange, I did manually test with the flag too lol.